airbyte-cdk/python/airbyte_cdk/sources/declarative/parsers/model_to_component_factory.py

```diff
@@ -764,18 +770,25 @@ def create_default_paginator(
             return PaginatorTestReadDecorator(paginator, self._limit_pages_fetched_per_slice)
         return paginator
 
-    def create_dpath_extractor(self, model: DpathExtractorModel, config: Config, **kwargs: Any) -> DpathExtractor:
-        decoder = self._create_component_from_model(model.decoder, config=config) if model.decoder else JsonDecoder(parameters={})
+    def create_dpath_extractor(
+        self, model: DpathExtractorModel, config: Config, decoder: Optional[Decoder] = None, **kwargs: Any
+    ) -> DpathExtractor:
+        if model.decoder:
```

Comment:
should we fail if there is both a model.decoder and a top-level decoder to avoid the ambiguity?

Reply:
left this for testing purposes only

> [!IMPORTANT]
Decoder has to be shared (be exactly the same object) across all components , because response is not reentrant safe where streamable=True

i changed order in `def create_simple_retriever` so we should set decoder on retriever level (json will be used if not set) and then pass this object to all subcomponents https://github.com/airbytehq/airbyte/blob/dc8e89e03db6af35058ca81eaae205837572ad15/airbyte-cdk/python/airbyte_cdk/sources/declarative/parsers/model_to_component_factory.py#L1097-L1102